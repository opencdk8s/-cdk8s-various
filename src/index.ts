import { ApiObject } from 'cdk8s';
import { Construct } from 'constructs';

export interface SecurityGroupPolicyOptions {
  readonly selector?: Record<string, string>;
  readonly securityGroupIds?: string[];
}

export class SecurityGroupPolicy extends Construct {
  constructor(scope: Construct, name: string, opts: SecurityGroupPolicyOptions) {
    super(scope, `${name}-sgp`);

    new ApiObject(this, name, {
      apiVersion: 'vpcresources.k8s.aws/v1beta1',
      kind: 'SecurityGroupPolicy',
      metadata: {
        name: name,
      },
      spec: {
        podSelector: {
          matchLabels: opts.selector,
        },
        securityGroups: {
          groupIds: opts.securityGroupIds,
        },
      },
    });
  }
}
