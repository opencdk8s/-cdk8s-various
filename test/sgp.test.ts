import { Chart, Testing } from 'cdk8s';
import { SecurityGroupPolicy } from '../src/index';

test('sgp', () => {

  const app = Testing.app();
  const chart = new Chart(app, 'test');

  new SecurityGroupPolicy(chart, 'test', {
    selector: {
      app: 'test',
    },
    securityGroupIds: ['sg-12345'],
  });


  expect(Testing.synth(chart)).toMatchSnapshot();
});
