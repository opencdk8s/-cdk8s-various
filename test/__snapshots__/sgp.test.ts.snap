// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sgp 1`] = `
Array [
  Object {
    "apiVersion": "vpcresources.k8s.aws/v1beta1",
    "kind": "SecurityGroupPolicy",
    "metadata": Object {
      "name": "test",
    },
    "spec": Object {
      "podSelector": Object {
        "matchLabels": Object {
          "app": "test",
        },
      },
      "securityGroups": Object {
        "groupIds": Array [
          "sg-12345",
        ],
      },
    },
  },
]
`;
